plugin.video.carnet-meduza/resources/lib/modules/control.py

```diff
@@ -0,0 +1,235 @@
+# -*- coding: utf-8 -*-
+
+'''
+    Exodus Add-on
+    Copyright (C) 2016 Exodus
+
+    This program is free software: you can redistribute it and/or modify
+    it under the terms of the GNU General Public License as published by
+    the Free Software Foundation, either version 3 of the License, or
+    (at your option) any later version.
+
+    This program is distributed in the hope that it will be useful,
+    but WITHOUT ANY WARRANTY; without even the implied warranty of
+    MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+    GNU General Public License for more details.
+
+    You should have received a copy of the GNU General Public License
+    along with this program.  If not, see <http://www.gnu.org/licenses/>.
+'''
+
+
+import os,xbmc,xbmcaddon,xbmcplugin,xbmcgui,xbmcvfs
+
+
+integer = 1
+
+lang = xbmcaddon.Addon().getLocalizedString
+
+setting = xbmcaddon.Addon().getSetting
+
+setSetting = xbmcaddon.Addon().setSetting
+
+addon = xbmcaddon.Addon
+
+addItem = xbmcplugin.addDirectoryItem
+
+item = xbmcgui.ListItem
+
+directory = xbmcplugin.endOfDirectory
+
+content = xbmcplugin.setContent
+
+property = xbmcplugin.setProperty
+
+addonInfo = xbmcaddon.Addon().getAddonInfo
+
+infoLabel = xbmc.getInfoLabel
+
+condVisibility = xbmc.getCondVisibility
+
+jsonrpc = xbmc.executeJSONRPC
+
+window = xbmcgui.Window(10000)
+
+dialog = xbmcgui.Dialog()
+
+progressDialog = xbmcgui.DialogProgress()
+
+windowDialog = xbmcgui.WindowDialog()
+
+button = xbmcgui.ControlButton
+
+image = xbmcgui.ControlImage
+
+keyboard = xbmc.Keyboard
+
+sleep = xbmc.sleep
+
+execute = xbmc.executebuiltin
+
+skin = xbmc.getSkinDir()
+
+player = xbmc.Player()
+
+playlist = xbmc.PlayList(xbmc.PLAYLIST_VIDEO)
+
+resolve = xbmcplugin.setResolvedUrl
+
+openFile = xbmcvfs.File
+
+makeFile = xbmcvfs.mkdir
+
+deleteFile = xbmcvfs.delete
+
+listDir = xbmcvfs.listdir
+
+transPath = xbmc.translatePath
+
+skinPath = xbmc.translatePath('special://skin/')
+
+addonPath = xbmc.translatePath(addonInfo('path'))
+
+dataPath = xbmc.translatePath(addonInfo('profile')).decode('utf-8')
+
+settingsFile = os.path.join(dataPath, 'settings.xml')
+
+viewsFile = os.path.join(dataPath, 'views.db')
+
+bookmarksFile = os.path.join(dataPath, 'bookmarks.db')
+
+providercacheFile = os.path.join(dataPath, 'providers.2.db')
+
+cachemetaFile = os.path.join(dataPath, 'metacache.db')
+
+metacacheFile = os.path.join(dataPath, 'meta.db')
+
+cacheFile = os.path.join(dataPath, 'cache.db')
+
+favouritesFile = os.path.join(dataPath, 'favourites.db')
+
+
+def addonIcon():
+    appearance = setting('appearance.1').lower()
+    if appearance in ['-', '']: return addonInfo('icon')
+    try: return os.path.join(xbmcaddon.Addon('script.exodus.artwork').getAddonInfo('path'), 'resources', 'media', appearance, 'icon.png')

```

Comment:
Wow, this DEFINITELY will not be allowed. I suggest you rewrite the entire module to sanitize.